packages/babel-parser/test/fixtures/annex-b/enabled/assignmenttargettype/input.js

```diff
@@ -0,0 +1,12 @@
+function async() {}
+
+async() = 1;
+async() += 1;
+async()++;
+++async();
+for (async() in [1]) { }
+for (async() of [1]) {}
```

Comment:
Could you also add a negative case for parenthesized expressions, such as `(f()) = 1`. Since the `AssignmentTargetType` will pass through the parentheses.